src/cursor/cur_backup_incr.c

```diff
@@ -186,8 +176,32 @@ __curbackup_incr_next(WT_CURSOR *cursor)
                 WT_ERR(WT_NOTFOUND);
             }
         }
-        __wt_cursor_set_key(cursor, cb->offset + cb->granularity * cb->bit_offset++,
-          cb->granularity, WT_BACKUP_RANGE);
+        /* We have initialized incremental information. */
+        start_bitoff = cb->bit_offset;
+        total_len = cb->granularity;
+        /* Look for the next chunk that had modifications.  */
+        while (cb->bit_offset < cb->nbits)
+            if (__bit_test(cb->bitstring.mem, cb->bit_offset)) {
+                /*
+                 * Care must be taken to leave the bit_offset field set to the next offset bit so
+                 * that the next call is set to the correct offset.
+                 */
+                start_bitoff = cb->bit_offset++;
+                if (F_ISSET(cb, WT_CURBACKUP_CONSOLIDATE)) {
+                    while (__bit_test(cb->bitstring.mem, cb->bit_offset++))
+                        total_len += cb->granularity;
+                }
+                break;
+            } else
+                ++cb->bit_offset;
+
+        /* We either have this object's incremental information or we're done. */
+        if (cb->bit_offset >= cb->nbits)
```

Comment:
Can this return `WT_NOTFOUND` when the final bit(s) in the map are set?  Unlike the previous version we increment past the last set bit before this check, rather than later in `wt_cursor_set_key()`.

Reply:
Good catch. I changed this to a boolean. Fixed.